var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -97,6 +97,7 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
         depends_on("magma@{0}:".format(magma_v), when="@{0}:+rocm".format(hiop_v))
 
     depends_on("cuda@11:", when="@develop:+cuda")
+    conflicts("+cusolver_lu", when="~cuda", msg="Cusolver requires CUDA")
 
```

Comment:
cuSolver LU refactorization needs KLU to get initial L and U factors and their sparsity patterns. 
```suggestion
    conflicts("+cusolver_lu", when="~cuda", msg="Cusolver LU refactorization requires CUDA")
    conflicts("+cusolver_lu", when="~suite-sparse", msg="Cusolver LU refactorization requires KLU from SuiteSparse library")

```